src/iterators.jl

```diff
@@ -41,7 +41,7 @@ struct CellIterator{dim,C,T,DH<:Union{AbstractDofHandler,Nothing}}
     dh::Union{DH,Nothing}
     celldofs::Vector{Int}
 
```

Comment:
Seems like the CellIterator will still have an Abstract celltype? 
This PR doesn't seem necessary for fixing that but would enable lifting the type instability higher up (i.e. making the creation of CellIterator type-stable also) !

Reply:
Haven't done anything on the `CellIterator` yet, I'd hope fixing the `CellIterator` for mixed grids would come naturally once we agree on a grid format.